### PR TITLE
Make constructors const

### PR DESCRIPTION
**Pull request by tcharding**
Audit the codebase for any function that starts with `/// Creates` and see if we can make it const. Inline them at the same time.

### CODE REVIEW

File: bitcoin/src/blockdata/witness.rs [hunk 1]
```diff
@@ -233,6 +233,7 @@ impl Encodable for Witness {
 
 impl Witness {
     /// Creates a new empty [`Witness`].
+    #[inline]
     pub fn new() -> Self { Witness::default() }
```
Comment (Kixunil) [inline bitcoin/src/blockdata/witness.rs:237]: I think this could be const too if you move the code from `default` here and call `new` from `default`.

**Review by Kixunil — APPROVED** [review]
ACK 0d6a8dea0443380c604148ed9cfeba02a216dc98

File: bitcoin/src/blockdata/witness.rs (hunk 1 above)
Comment (apoelstra) [reply bitcoin/src/blockdata/witness.rs:237]: I believe so. Looks like Vec::new has been const since 1.39.

Comment (tcharding) [reply bitcoin/src/blockdata/witness.rs:237]: Nice, thanks!

**Comment by tcharding** [comment]
Bump please @apoelstra

**Comment by Kixunil** [comment]
Nit: your commit message uses backwards slashes.

**Comment by apoelstra** [comment]
Oops, I did not see this PR.

Can you rebase to get the CI fixes? My local CI is complaining about them.

You can also fix Kix's nit while you're doing so.

**Comment by tcharding** [comment]
> Nit: your commit message uses backwards slashes.

Lol, its my first time typing.

**Comment by tcharding** [comment]
Needs #2499

**Review by apoelstra — DISMISSED** [review]
ACK f75e39e475e948d91087e0b359957bb34ef980ee

File: bitcoin/src/blockdata/witness.rs (hunk 1 above)
Comment (Kixunil) [reply bitcoin/src/blockdata/witness.rs:237]: Did you mean to make this `const`?

**Review by Kixunil — APPROVED** [review]
ACK f75e39e475e948d91087e0b359957bb34ef980ee

**Comment by tcharding** [comment]
Force push makes `Witnesss:new` const.

**Review by Kixunil — APPROVED** [review]
ACK ac88bc03fda57f55e39e59a300a20dc364173b39

**Review by apoelstra — APPROVED** [review]
ACK ac88bc03fda57f55e39e59a300a20dc364173b39

**Pull request merged**